Fix unverified exception in JSONPPathEntryTypeCall

hyjorc1 opened the pull request:
#### 6f10652c4e36c36dbb79c8f9aaed8b51bcdcb8ae
<pre>
Fix unverified exception in JSONPPathEntryTypeCall
<a href="https://bugs.webkit.org/show_bug.cgi?id=243674">https://bugs.webkit.org/show_bug.cgi?id=243674</a>
rdar://98326554

Reviewed by Alexey Shvayka.

Since hasProperty might throw an exception, we need to handle the exception in both
branches before proceeding.

* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::Interpreter::executeProgram):

Canonical link: <a href="https://commits.webkit.org/253233@main">https://commits.webkit.org/253233@main</a>
</pre>

shvaikalesh reviewed (approved):
Alternatively we could do:

```cpp
bool hasProperty = scope->hasProperty(globalObject, ident);
RETURN_IF_EXCEPTION(throwScope, { });
if (hasProperty) {
    // ...
}
```

MenloDorian commented on Source/JavaScriptCore/interpreter/Interpreter.cpp:
```diff
@@ -910,6 +910,7 @@ JSValue Interpreter::executeProgram(const SourceCode& source, JSGlobalObject*, J
                         return throwException(globalObject, throwScope, createTDZError(globalObject));
                     baseObject = scope;
                 }
+                RETURN_IF_EXCEPTION(throwScope, JSValue());
```
The better way to this is:
```
bool hasProperty = scope->hasProperty(globalObject, ident));
RETURN_IF_EXCEPTION(throwScope, JSValue());
if (hasProperty) {
    ...
```

MenloDorian replied:
I see that Alexey just suggested the same thing.

hyjorc1 replied:
Updated.

MenloDorian reviewed (approved):
r=me

hyjorc1 committed:
Fix unverified exception in JSONPPathEntryTypeCall
https://bugs.webkit.org/show_bug.cgi?id=243674
rdar://98326554

Reviewed by Alexey Shvayka.

Since hasProperty might throw an exception, we need to handle the exception in both
branches before proceeding.

* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::Interpreter::executeProgram):

Canonical link: https://commits.webkit.org/253233@main
